### ISSUE TITLE
Inject certs in `docker` step type

### ISSUE THREAD

**Issue by jilleJr** [iver-wharf/wharf-cmd#123]
Need to inject certificates into the Docker step type, somewhere inside `pkg/worker/k8spodtemplating.go`.

This is a weird thing we've done previously to add our internal self-signed certs, but we cannot forego it yet as we have a lot of stuff depending on it.

The current implementation from our Jenkinsfiles looks like this:

```sh
if [ "${params.DOCKERFILE_APPEND_CERT}" == "true" ]
then

cert_path=`pwd`/${params.DOCKERFILE_CONTEXT}/root.crt
cat <<EOF > root.crt
${params.ROOT_CERT}
EOF
cp root.crt \$cert_path
echo -e "\n" >> `pwd`/${params.DOCKERFILE_PATH}
echo COPY ./root.crt /usr/local/share/ca-certificates/root.crt >> `pwd`/${params.DOCKERFILE_PATH}
echo "RUN mkdir -p /etc/ssl/certs/ && touch /etc/ssl/certs/ca-certificates.crt && cat /usr/local/share/ca-certificates/root.crt >> /etc/ssl/certs/ca-certificates.crt"  >> `pwd`/${params.DOCKERFILE_PATH}

fi
```

https://github.com/iver-wharf/wharf-jenkinsfiles/blob/2811389f00a4335187e2f5b0a1c356eddac99494/build.git/kaniko/Jenkinsfile#L110-L125

In other words:

1. Copy the certificate into the build context
2. Add the following to the Dockerfile:

   ```dockerfile
   COPY ./root.crt /usr/local/share/ca-certificates/root.crt
   RUN mkdir -p /etc/ssl/certs/ \
       && touch /etc/ssl/certs/ca-certificates.crt \
       && cat /usr/local/share/ca-certificates/root.crt >> /etc/ssl/certs/ca-certificates.crt
   ```

There are different ways to accomplish this from wharf-cmd's perspective, but the preferred way we could do it would be without adding Bash code to the container, as the kaniko image doesn't have a shell installed.

**Comment by jilleJr** [comment]
Idea:

- After copy tar file, copy the modified Dockerfile
- Mount cert into init container and then exec something like `cp /mnt/cert/root.crt /mnt/repo/root.crt`

**Comment by Alexamakans** [comment]
> Idea:
> 
> * After copy tar file, copy the modified Dockerfile
> * Mount cert into init container and then exec something like `cp /mnt/cert/root.crt /mnt/repo/root.crt`

Giving this a go 👍🏻

**Comment by jilleJr** [comment]
As we have full control over the streaming data in Go, I think the implementation would be quite simple. No need to edit the Dockerfile on disk, just do a `io.Copy` on the file's content over through the stream, and then pipe those commands from above as well. On the receiving end just run something like `tee path/to/Dockerfile`, and let it receive the modified Dockerfile through STDIN
